### PR TITLE
[Tabs] Add support for `selectedImage` from UITabBarItem.

### LINKED ISSUE #7798: [Tabs] Internal issue: b/136938054
This was filed as an internal issue. If you are a Googler, please visit [b/136938054](http://b/136938054) for more details.

---
#### Internal data
- Associated internal bug: [b/136938054](http://b/136938054)

### PR DESCRIPTION
**Pull request by romoore**
MDCTabBarView needs to support `selectedImage` from UITabBarItem. Some clients will want to use different images for selected/unselected states.

Closes #7798

### CODE REVIEW

File: components/Tabs/src/TabBarView/MDCTabBarView.m [hunk 1]
```diff
@@ -451,7 +458,13 @@ - (void)observeValueForKeyPath:(NSString *)keyPath
       newValue = nil;
     }
     if ([keyPath isEqualToString:kImageKeyPath]) {
-      tabBarItemView.iconImageView.image = newValue;
+      tabBarItemView.image = newValue;
+      [tabBarItemView invalidateIntrinsicContentSize];
```
Comment (codeman7) [inline components/Tabs/src/TabBarView/MDCTabBarView.m:462]: Can we clean this up either here or in a follow up. I see these four lines repeated 3 times here.

File: components/Tabs/src/TabBarView/private/MDCTabBarViewItemView.m [hunk 2]
```diff
@@ -248,6 +271,21 @@ - (NSString *)accessibilityLabel {
...
+    self->_selected = selected;
+    if (selected) {
+      self.iconImageView.image = self.selectedImage ?: self.image;
+    } else {
+      self.iconImageView.image = self.image;
+    }
+  };
+
+  if (animated) {
+    [UIView animateWithDuration:0.3 animations:animationBlock];
```
Comment (codeman7) [inline components/Tabs/src/TabBarView/private/MDCTabBarViewItemView.m:284]: Where is this `0.3` coming from?

**Review by codeman7 — APPROVED** [review]

File: components/Tabs/src/TabBarView/private/MDCTabBarViewItemView.m (hunk 2 above)
Comment (romoore) [reply components/Tabs/src/TabBarView/private/MDCTabBarViewItemView.m:284]: From MDCTabBarView.

File: components/Tabs/src/TabBarView/MDCTabBarView.m (hunk 1 above)
Comment (romoore) [reply components/Tabs/src/TabBarView/MDCTabBarView.m:462]: Sure. Will post a follow-up PR after this merges.

**Pull request merged**